src/app/core/market/proposals-notifier/proposals-notifications.service.ts

```diff
@@ -68,6 +71,9 @@ export class ProposalsNotificationsService implements OnDestroy {
             }
             if (needUpdating) {
               this.notifcationTimestamp = Date.now();
+
+              // restrict the notification for the unseen proposal at time time of GUI started.
+              localStorage.setItem(this.storageKeys.timestamp_notifcation, String(this.notifcationTimestamp));
```

Comment:
This line can be moved into the storeLastViewedProposalTimestamp() method (which can also be renamed if you feel up to it :) ). 
Rather keep the saving/writing of data in the same method, and leave this function to its purpose of data processing.

Reply:
@zaSmilingIdiot 
As I mentioned before in our conversation we need to update the timestamp_notificaiton, in that case as well if the user does not go to the proposal page.

If I put that code in the storeLastViewedProposalTimestamp()
then timestapm_notification only in that case when I go to the proposal page, as storeLastViewedProposalTimestamp() called in that case only with the help if viewingProposals().

So It's hard to maintain for first time notification (afterwords GUI started). if not updated the timestamp in the storage after notification pops up.